src/providers/AccountProvider.tsx

```diff
@@ -20,59 +20,60 @@ export interface Account {
 }
 
 export interface AccountProviderContext {
-  account?: Account | null;
-  accounts?: Account[];
+  accounts: Account[];
   createAccount: () => void;
   selectedAccount?: Account;
   selectAccount: (network: Account) => void;
 }
 
 const AccountsContext = React.createContext<AccountProviderContext>({
+  accounts: [],
   createAccount() {},
-  selectAccount(n: Account) {},
+  selectAccount(_account: Account) {},
+  selectedAccount: undefined,
 });
 
 function AccountProvider(props: {children: ReactNode}) {
   const {children} = props;
   const [accounts, setAccounts] = useState<Account[]>([]);
   const [selectedAccount, setSelectedAccount] = useState<Account>(accounts[0]);
 
-  const [account, setAccount] = useState<Account | null>();
-
-  const createAccount = () => {
+  function createAccount() {
     const {publicKey, secretKey} = Keypair.generate();
     const name = elipsify(publicKey.toBase58());
 
-    setAccounts([
-      ...accounts,
+    setAccounts(currentAccounts => [
+      ...currentAccounts,
       {
         id: publicKey.toBase58(),
         name,
         publicKey: publicKey,
         secretKey: secretKey,
       },
     ]);
-  };
-  const selectAccount = (network: Account) => {
-    console.log(`Select Account: ${network?.name}`);
-    setSelectedAccount(network);
-    // setAccount(() => new Account(clusterApiUrl(network.endpoint)));
-  };
+  }
+
+  function selectAccount(account: Account) {
```

Comment:
This function doesn't need to be memoized through `useCallback` because its only dependency, `setSelectedAccount`, is stable.